statesync/snapshots.go

```diff
@@ -177,12 +177,41 @@ func (p *snapshotPool) Ranked() []*snapshot {
 	p.Lock()
 	defer p.Unlock()
 
-	candidates := make([]*snapshot, 0, len(p.snapshots))
+	if len(p.snapshots) == 0 {
+		return []*snapshot{}
+	}
+
+	numPeers := make([]int, 0, len(p.snapshots))
 	for key := range p.snapshots {
-		candidates = append(candidates, p.snapshots[key])
+		numPeers = append(numPeers, len(p.snapshotPeers[key]))
+	}
+	sort.Ints(numPeers)
+	median := len(numPeers) / 2
+	if len(numPeers)%2 == 0 {
+		median = (numPeers[median-1] + numPeers[median]) / 2
+	} else {
+		median = numPeers[median]
 	}
 
-	sort.Slice(candidates, func(i, j int) bool {
+	commonCandidates := make([]*snapshot, 0, len(p.snapshots)/2)
+	uncommonCandidates := make([]*snapshot, 0, len(p.snapshots)/2)
```

Comment:
do we even need these lol?

Reply:
unsure, but it doesn't really hurt much?